db/error_handler.cc

```diff
@@ -328,7 +336,14 @@ const Status& ErrorHandler::SetBGError(const Status& bg_err,
 
   // Allow some error specific overrides
   if (new_bg_err == Status::NoSpace()) {
-    new_bg_err = OverrideNoSpaceError(new_bg_err, &auto_recovery);
+    new_bg_err = OverrideNoSpaceError(new_bg_err, reason, &auto_recovery);
+  }
+
+  if ((!db_options_.max_bgerror_resume_count || !auto_recovery) &&
```

Comment:
How about compaction? Compaction do not do auto recovery and it just reschedule by itself. We set it to soft error. Should it be hard error?

Reply:
If the user has disabled auto recovery, I think we should set it to hard error even for compaction. Otherwise, too many pending compactions could also lead to a write stall.